dns/resolver.go

```diff
@@ -391,3 +396,14 @@ func NewHTTPSResolver(sd transport.StreamDialer, resolverAddr string, url string
 		return &msg, nil
 	})
 }
+
+func wrappErrors(err1, err2 error) error {
```

Comment:
@fortuna we need to use this function instead of `errors.Join()` since `errors.Join()` breaks the wrapped chain of errors. when we are unwrapping the errors  with `unwrapAll()` in `makeErrorRecord` in `test_connectivity.go`, the errors don't get fully unwrapped and can leak unwanted error messages that can contain sensitive information.